### PR TITLE
[#158179629] feat: return sender_allowed with limited profile

### PR DESCRIPTION
**Pull request by gunzip**
we do not return the sender_allowed flag with full profile as with a full profile we got the whole array with blocked services

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#158179629](https://www.pivotaltracker.com/story/show/158179629): Come CIT, quando un servizio vuole inviarmi una comunicazione, voglio che il servizio sappia che ho fatto optout dalle sue comunicazioni

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: api/definitions.yaml [hunk 1]
```diff
@@ -253,8 +253,13 @@ LimitedProfile:
     attributes.
   type: object
   properties:
+    sender_allowed:
+      type: boolean
+      default: true
```
Comment (cloudify) [inline api/definitions.yaml]: what does the default do in our logic?

Comment (cloudify) [reply api/definitions.yaml]: also worth adding a description since this is shared with the PAs

File: lib/controllers/profiles.ts [hunk 2]
```diff
@@ -65,8 +72,30 @@ function toExtendedProfile(profile: RetrievedProfile): ExtendedProfile {
...
+  return (
+    blockedInboxOrChannels === undefined ||
+    blockedInboxOrChannels[serviceId] === undefined ||
+    !blockedInboxOrChannels[serviceId].has(BlockedInboxOrChannelEnum.INBOX)
+  );
+}
+
+function toLimitedProfile(
+  profile: RetrievedProfile,
+  senderAllowed: boolean
```
Comment (cloudify) [inline lib/controllers/profiles.ts]: `isSenderAllowed`

Comment (gunzip) [reply lib/controllers/profiles.ts]: cannot shadow function name

File: api/definitions.yaml (hunk 1 above)
Comment (gunzip) [reply api/definitions.yaml]: removed default and added description

**Pull request merged**